sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -136,56 +151,62 @@ public static Read read() {
 
       abstract Builder setInitialPosition(StartingPoint startingPoint);
 
-      abstract Builder setClientProvider(KinesisClientProvider clientProvider);
+      abstract Builder setAWSClientsProvider(AWSClientsProvider clientProvider);
 
       abstract Builder setMaxNumRecords(int maxNumRecords);
 
       abstract Builder setMaxReadTime(Duration maxReadTime);
 
+      abstract Builder setAllowedRecordLateness(Duration allowedRecordLateness);
+
       abstract Read build();
     }
 
     /**
-     * Specify reading from streamName at some initial position.
+     * Specify reading from streamName.
      */
-    public Read from(String streamName, InitialPositionInStream initialPosition) {
-      return toBuilder()
-          .setStreamName(streamName)
-          .setInitialPosition(
-              new StartingPoint(checkNotNull(initialPosition, "initialPosition")))
+    public Read withStreamName(String streamName) {
```

Comment:
Oh right, I didn't notice it was marked Experimental. I guess then it's fine to proceed with the rest of the changes in this PR; I defer for further review to JB.